core/drand_beacon.go

```diff
@@ -360,6 +361,37 @@ func (bp *BeaconProcess) isFreshRun() bool {
 	return errG != nil || errS != nil
 }
 
+// getChainHash return the beaconID of that beaconProcess, if set
+func (bp *BeaconProcess) getBeaconID() string {
+	return bp.beaconID
+}
+
+// getChainHash return the HashChain in hex format as a string
+func (bp *BeaconProcess) getChainHash() []byte {
+	if len(bp.chainHash) == 0 && bp.group != nil {
+		// we only lock the state in the case it's not yet set and we set it since we only set it in Load() otherwise
+		bp.state.Lock()
+		defer bp.state.Unlock()
+		info := chain.NewChainInfo(bp.group)
+		bp.chainHash = info.Hash()
+	}
+
+	return bp.chainHash
+}
+
+func (bp *BeaconProcess) newMetadata() *common.Metadata {
+	metadata := common.NewMetadata(bp.version.ToProto())
+	metadata.BeaconID = bp.getBeaconID()
+
+	if hash := bp.getChainHash(); len(hash) > 0 {
+		metadata.ChainHash = hash
+	} else {
+		bp.log.Errorw("No chain hash set yet, waiting for DKG")
```

Comment:
It's not an error - it's how it is, no chain hash to set if dkg hasn't been run, but the metadata is still valid